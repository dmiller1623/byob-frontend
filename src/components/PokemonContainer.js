import Pokemoncard from './PokemonCard';
import React from 'react';

export const PokemonContainer = ({ pokemon }) => {
  const displayPokemon = pokemon.map((poke, index) => {
    return <Pokemoncard 
      {...poke}
      key={index}
  />
  })
  return(
    <div>
      {displayPokemon}
    </div>
  )
}

