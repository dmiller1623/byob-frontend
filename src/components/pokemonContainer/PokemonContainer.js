import Pokemoncard from '../pokemonCard/PokemonCard';
import React from 'react';
import './pokemonContainer.css';
import rightArrow from '../../images/right-arrow.svg';
import leftArrow from '../../images/left-arrow.svg';
import PropTypes from 'prop-types';


export const PokemonContainer = ({ 
  pokemon,
  counter,
  increment,
  decrement,
  selectedPokemon,
  addToPokemonTeam,
  deleteFromTeam,
  trainers,
  postTeam,
  getCurrentTrainer
}) => {
  
  const displayPokemon = pokemon.map((poke, index) => {
    return <Pokemoncard 
      {...poke}
      key={index}
      addToPokemonTeam={addToPokemonTeam}
    />;
  });

  return (
    <div className='poke-container'>
      <div className='poke-card-container'>
        <img  alt='left button' className='carousel' src={leftArrow} onClick={() => decrement()}/>
        {displayPokemon[counter]}
        <img  alt='right button' className='carousel' src={rightArrow} onClick={() => increment()}/>
      </div>
      <div className='add-poke-section'>
        <select className='add-poke-buttons' onChange={(event) => getCurrentTrainer(event)}>
          <option>Choose Trainer</option>
          {trainers.map(trainer => {
            return <option key={trainer.id} value={trainer.name}>{trainer.name}</option>;
          })}
        </select>
        <button className='add-poke-buttons' onClick={postTeam}>save team</button>
      </div>
      <div className='main-team-display'>
        {selectedPokemon.map((pokemon, index) => {
          return <div key={index}>
            <img className='selected-poke' alt='selected pokemon' key={index} onClick={() => deleteFromTeam(index)} src={pokemon}/>
          </div>;
        })}
      </div>
      {selectedPokemon.length ?  <p className='remove-tag'>Click Icons to Remove</p> : <p></p>}
    </div>
  );
};

PokemonContainer.propTypes = {
  pokemon: PropTypes.array,
  counter: PropTypes.number,
  increment: PropTypes.func,
  decrement: PropTypes.func,
  selectedPokemon: PropTypes.array,
  addToPokemonTeam: PropTypes.func,
  deleteFromTeam: PropTypes.func,
  trainers: PropTypes.array,
  postTeam: PropTypes.func,
  getCurrentTrainer: PropTypes.func
};