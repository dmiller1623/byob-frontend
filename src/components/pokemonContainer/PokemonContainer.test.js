import React from 'react';
import { shallow } from 'enzyme'; 
import { PokemonContainer } from './PokemonContainer.js';

describe('TrainerSideBar', () => {
  let wrapper;
  let pokemonMock = [];
  let selectedPokemonMock = [];
  let trainersMock = [];

  it('should match the snapshot', () => {
    wrapper = shallow(<PokemonContainer pokemon={pokemonMock} trainers={trainersMock} selectedPokemon={selectedPokemonMock} />);
    expect(wrapper).toMatchSnapshot();
  });
});