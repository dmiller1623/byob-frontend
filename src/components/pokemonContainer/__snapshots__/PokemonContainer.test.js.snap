// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainerSideBar should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokemonContainer
    pokemon={Array []}
    selectedPokemon={Array []}
    trainers={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="poke-card-container"
        >
          <img
            alt="left button"
            className="carousel"
            onClick={[Function]}
            src="left-arrow.svg"
          />
          <img
            alt="right button"
            className="carousel"
            onClick={[Function]}
            src="right-arrow.svg"
          />
        </div>,
        <div
          className="add-poke-section"
        >
          <select
            className="add-poke-buttons"
            onChange={[Function]}
          >
            <option>
              Choose Trainer
            </option>
          </select>
          <button
            className="add-poke-buttons"
          >
            save team
          </button>
        </div>,
        <div
          className="main-team-display"
        />,
        <p />,
      ],
      "className": "poke-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="left button"
              className="carousel"
              onClick={[Function]}
              src="left-arrow.svg"
            />,
            undefined,
            <img
              alt="right button"
              className="carousel"
              onClick={[Function]}
              src="right-arrow.svg"
            />,
          ],
          "className": "poke-card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "left button",
              "className": "carousel",
              "onClick": [Function],
              "src": "left-arrow.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "right button",
              "className": "carousel",
              "onClick": [Function],
              "src": "right-arrow.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              className="add-poke-buttons"
              onChange={[Function]}
            >
              <option>
                Choose Trainer
              </option>
            </select>,
            <button
              className="add-poke-buttons"
            >
              save team
            </button>,
          ],
          "className": "add-poke-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option>
                  Choose Trainer
                </option>,
                Array [],
              ],
              "className": "add-poke-buttons",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Choose Trainer",
                },
                "ref": null,
                "rendered": "Choose Trainer",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "save team",
              "className": "add-poke-buttons",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "save team",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "main-team-display",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="poke-card-container"
          >
            <img
              alt="left button"
              className="carousel"
              onClick={[Function]}
              src="left-arrow.svg"
            />
            <img
              alt="right button"
              className="carousel"
              onClick={[Function]}
              src="right-arrow.svg"
            />
          </div>,
          <div
            className="add-poke-section"
          >
            <select
              className="add-poke-buttons"
              onChange={[Function]}
            >
              <option>
                Choose Trainer
              </option>
            </select>
            <button
              className="add-poke-buttons"
            >
              save team
            </button>
          </div>,
          <div
            className="main-team-display"
          />,
          <p />,
        ],
        "className": "poke-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="left button"
                className="carousel"
                onClick={[Function]}
                src="left-arrow.svg"
              />,
              undefined,
              <img
                alt="right button"
                className="carousel"
                onClick={[Function]}
                src="right-arrow.svg"
              />,
            ],
            "className": "poke-card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "left button",
                "className": "carousel",
                "onClick": [Function],
                "src": "left-arrow.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "right button",
                "className": "carousel",
                "onClick": [Function],
                "src": "right-arrow.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <select
                className="add-poke-buttons"
                onChange={[Function]}
              >
                <option>
                  Choose Trainer
                </option>
              </select>,
              <button
                className="add-poke-buttons"
              >
                save team
              </button>,
            ],
            "className": "add-poke-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option>
                    Choose Trainer
                  </option>,
                  Array [],
                ],
                "className": "add-poke-buttons",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Choose Trainer",
                  },
                  "ref": null,
                  "rendered": "Choose Trainer",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "save team",
                "className": "add-poke-buttons",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "save team",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "main-team-display",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
